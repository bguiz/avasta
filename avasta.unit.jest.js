'use strict';

const avasta = require('./avasta.js');

describe('[avasta]', () => {
	let res;
	let resStatus;
	let resJson;
 	let testCodes;
 	let testAvasta;

	beforeAll(() => {
		testCodes = require('./avasta.unit.jest.json');
		testAvasta = avasta(testCodes);
		resJson = jest.fn();
		resStatus = jest.fn();
		res = {
			json: resJson,
			status: resStatus,
		};
		resJson.mockImplementation(() => {
			return res;
		});
		resStatus.mockImplementation(() => {
			return res;
		});
	});

	describe('[express]', () => {

		it('handle name and message', () => {
			testAvasta.expressJsonResponse(res, '1st 400 error', 'custom message');
			expect({
				status: resStatus.mock.calls[resStatus.mock.calls.length - 1][0],
				json: resJson.mock.calls[resJson.mock.calls.length - 1][0],
			}).toMatchSnapshot();
		});

		it('handle name without message', () => {
			testAvasta.expressJsonResponse(res, '1st 400 error');
			expect({
				status: resStatus.mock.calls[resStatus.mock.calls.length - 1][0],
				json: resJson.mock.calls[resJson.mock.calls.length - 1][0],
			}).toMatchSnapshot();
		});

		it('handle undefined name', () => {
			testAvasta.expressJsonResponse(res, 'this is not a known name');
			expect({
				status: resStatus.mock.calls[resStatus.mock.calls.length - 1][0],
				json: resJson.mock.calls[resJson.mock.calls.length - 1][0],
			}).toMatchSnapshot();
		});

	});

	describe('[schwag]', () => {

		it('handle name and message', () => {
			const output = {};
			testAvasta.schwagResponse(output, '1st 400 error', 'custom message');
			expect(output).toMatchSnapshot();
		});

		it('handle name without message', () => {
			const output = {};
			testAvasta.schwagResponse(output, '1st 400 error');
			expect(output).toMatchSnapshot();
		});

		it('handle undefined name', () => {
			const output = {};
			testAvasta.schwagResponse(output, 'this is not a known name');
			expect(output).toMatchSnapshot();
		});

	});

});
