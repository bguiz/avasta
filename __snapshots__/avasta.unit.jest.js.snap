// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[avasta] [express] handle name and message 1`] = `
{
  "json": {
    "code": 400000,
    "message": "custom message"
  },
  "status": 400
}
`;

exports[`[avasta] [express] handle name without message 1`] = `
{
  "json": {
    "code": 400000
  },
  "status": 400
}
`;

exports[`[avasta] [express] handle undefined name 1`] = `
{
  "json": {
    "code": 599999
  },
  "status": 599
}
`;

exports[`[avasta] [schwag] handle name and message 1`] = `
{
  "body": {
    "code": 400000,
    "message": "custom message"
  },
  "status": 400
}
`;

exports[`[avasta] [schwag] handle name without message 1`] = `
{
  "body": {
    "code": 400000
  },
  "status": 400
}
`;

exports[`[avasta] [schwag] handle undefined name 1`] = `
{
  "body": {
    "code": 599999
  },
  "status": 599
}
`;
